modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/service/persistence/impl/KBArticleFinderImpl.java

```diff
@@ -127,6 +234,83 @@ protected String replaceWorkflowStatus(String sql, int[] status) {
 		return StringUtil.replace(sql, "[$WORKFLOW_STATUS$]", sb.toString());
 	}
 
+	private Predicate _getKeywordsPredicate(
+		Expression<String> expression, String[] keywords) {
+
+		expression = DSLFunctionFactoryUtil.lower(expression);
+
+		Predicate keywordsPredicate = null;
+
+		for (String keyword : keywords) {
+			if (keyword == null) {
+				continue;
+			}
+
+			Predicate keywordPredicate = expression.like(keyword);
+
+			if (keywordsPredicate == null) {
+				keywordsPredicate = keywordPredicate;
+			}
+			else {
+				keywordsPredicate = keywordsPredicate.or(keywordPredicate);
+			}
+		}
+
+		return keywordsPredicate;
+	}
+
+	private OrderByStep _getOrderByStep(
+		FromStep fromStep, long groupId, String keywords, int status,
+		boolean inlineSQLHelper) {
+
+		Predicate wherePredicate = KBArticleTable.INSTANCE.groupId.eq(groupId);
+
+		if (inlineSQLHelper) {
+			wherePredicate = Predicate.withParentheses(
+				wherePredicate.and(
+					InlineSQLHelperUtil.getPermissionWherePredicate(
+						KBArticle.class, KBArticleTable.INSTANCE.kbArticleId,
+						groupId)));
+		}
+
+		if (Validator.isNotNull(keywords)) {
+			String[] keywordsArray = _customSQL.keywords(keywords, true);
+
+			Predicate keywordsPredicate = Predicate.withParentheses(
+				_getKeywordsPredicate(
+					KBArticleTable.INSTANCE.title, keywordsArray
+				).or(
+					_getKeywordsPredicate(
+						DSLFunctionFactoryUtil.castClobText(
+							KBArticleTable.INSTANCE.content),
+						keywordsArray)
+				));
+
+			wherePredicate = wherePredicate.and(keywordsPredicate);
+		}
+
+		if (status != WorkflowConstants.STATUS_ANY) {
+			wherePredicate = wherePredicate.and(
+				KBArticleTable.INSTANCE.status.eq(status));
+		}
+
+		if (status == WorkflowConstants.STATUS_APPROVED) {
+			wherePredicate = wherePredicate.and(
+				KBArticleTable.INSTANCE.main.eq(Boolean.TRUE));
+		}
+
+		if (status == WorkflowConstants.STATUS_ANY) {
+			wherePredicate = wherePredicate.and(
+				KBArticleTable.INSTANCE.latest.eq(Boolean.TRUE));
+		}
```

Comment:
This looks wrong. In particular, the conditional that checks for "approved" will never be executed.